src/main/scala/mesosphere/marathon/api/akkahttp/v2/TasksController.scala

```diff
@@ -0,0 +1,105 @@
+import mesosphere.marathon.plugin.auth.{ Authenticator, _ }
+import play.api.libs.json.Json
+import mesosphere.marathon.raml.{ AnyToRaml, EnrichedTasksList, Writes }
+import mesosphere.marathon.state.PathId
+
+import scala.async.Async._
+import scala.concurrent.{ ExecutionContext, Future }
+
+class TasksController(instanceTracker: InstanceTracker, groupManager: GroupManager, healthCheckManager: HealthCheckManager)(
+    implicit
+    val actorSystem: ActorSystem,
+    val executionContext: ExecutionContext,
+    val authenticator: Authenticator,
+    val authorizer: Authorizer
+) extends Controller with StrictLogging {
+
+  import mesosphere.marathon.api.akkahttp.Directives._
+  import mesosphere.marathon.api.akkahttp.EntityMarshallers._
+  import mesosphere.marathon.raml.EnrichedTaskConversion._
+
+  override val route = {
+    get {
+      pathEndOrSingleSlash {
+        listTasks()
+      }
+    }
+  }
+
+  private def listTasks(): Route = {
+    authenticated.apply { implicit identity =>
+      authorized(ViewResource, AuthorizedResource.SystemConfig).apply {
+        parameters("status".?, "status[]".as[String].*) { (statusParameter, statusesParameter) =>
+          val statuses = statusParameter.map(s => Seq(s) ++ statusesParameter).getOrElse(Seq[String](statusesParameter.toSeq: _*))
+          onSuccess(enrichedTasks(statuses)) { tasks =>
+            complete(TasksList(tasks).toRaml)
+          }
+        }
+      }
+    }
+  }
+
+  @SuppressWarnings(Array("all")) // async/await
+  private def enrichedTasks(statuses: Seq[String]): Future[Seq[EnrichedTask]] = async {
+    val conditionSet: Set[Condition] = statuses.flatMap(toTaskState)(collection.breakOut)
+
+    val instancesBySpec = await(instanceTracker.instancesBySpec)
+
+    val instances: Iterable[(PathId, Instance)] = instancesBySpec.instancesMap.values.flatMap { appTasks =>
+      appTasks.instances.map(i => appTasks.specId -> i)
+    }
+    val appIds: Set[PathId] = instancesBySpec.allSpecIdsWithInstances
+
+    val appToPorts: Map[PathId, Seq[Int]] = groupManager.apps(appIds).map {
+      case (appId, app) => appId -> app.map(_.servicePorts).getOrElse(Nil)
+    }
+
+    val instancesHealth = await(
+      Future.sequence(appIds.map { appId =>
+        healthCheckManager.statuses(appId)
+      })).foldLeft(Map[Id, Seq[Health]]())(_ ++ _)
+
+    val enrichedTasks: Iterable[Iterable[EnrichedTask]] = for {
+      (appId, instance) <- instances
+      if conditionSet.isEmpty || conditionSet(instance.state.condition)
+      tasks = instance.tasksMap.values
+    } yield {
+      tasks.map { task =>
+        EnrichedTask(
+          appId,
+          task,
+          instance.agentInfo,
+          instancesHealth.getOrElse(instance.instanceId, Nil),
+          appToPorts.getOrElse(appId, Nil)
+        )
+      }
+    }
+    Seq(enrichedTasks.flatten.toSeq: _*)
+  }
+
+  private def toTaskState(state: String): Option[Condition] = state.toLowerCase match {
```

Comment:
Don't we have a mapping already? Also, I think it should be `toCondition`. @unterstein introduced `Condition`. He probably can explain the difference better. We should also document it then.

Reply:
for some reason, this endpoint works only with two conditions (running and staging), rest is ignored. So if we want to keep that, we probably need this... I'll look into it more

Reply:
Btw. I found that status[] was introduced here https://github.com/mesosphere/marathon/issues/476